api/client/alpn_conn_upgrade.go

```diff
@@ -124,7 +125,7 @@ func OverwriteALPNConnUpgradeRequirementByEnv(addr string) (bool, bool) {
 		return false, false
 	}
 	result := isALPNConnUpgradeRequiredByEnv(addr, envValue)
-	logrus.WithField(defaults.TLSRoutingConnUpgradeEnvVar, envValue).Debugf("ALPN connection upgrade required for %q: %v.", addr, result)
+	slog.DebugContext(context.TODO(), "Determining if ALPN connection upgrade is explicitly forced due to environment variables.", defaults.TLSRoutingConnUpgradeEnvVar, envValue, "address", addr, "upgrade_required", result)
```

Comment:
Maybe move to context.Background()? There's no follow up for this, is there?

Same for others.